frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -902,27 +925,50 @@ COPY Dockerfile${foo} .
 `)
 	checkLinterWarnings(t, sb, &lintTestParams{Dockerfile: dockerfile})
 
-	dockerfile = []byte(`
+	dockerfile = []byte(integration.UnixOrWindows(
+		`
 FROM alpine AS base
 ARG foo=Dockerfile
 
 FROM base
 COPY $foo .
-`)
+`,
+		`
+FROM nanoserver AS base
```

Comment:
For these Dockerfiles that only differ by the base image define `baseImage := integration.UnixOrWindow()` instead.

Reply:
I'd thought `fmt.Sprintf` will be a little ugly but let me fix that...

Reply:
Can also `` + integration.UnixOrWindow() + `` if that is cleaner

Reply:
Coz some come in between, might be better to use `fmt.Sprintf`.  I've made here, and the other instances in the PR. PTAL.